frontend/src/modules/common/form-fields/slug-field.tsx

```diff
@@ -0,0 +1,69 @@
+import { useMutation } from '@tanstack/react-query';
+import type { Control } from 'react-hook-form';
+import { useFormContext, useWatch } from 'react-hook-form';
+import { checkSlugAvailable } from '~/api/general';
+import InputFormField from './input';
+import { useEffect } from 'react';
+
+interface SlugFieldProps {
+  // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+  control: Control<any>;
+  name: string;
+  label: string;
+  required?: boolean;
+  onFocus?: () => void;
+  description?: string;
+  errorMessage?: string;
+  subComponent?: React.ReactNode;
+  previousSlug?: string;
+}
+
+export const SlugField = ({
```

Comment:
the name should be SlugFormField in my opinion. Its more consistent with what we already have